ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/TrailingCommaRuleTest.kt

```diff
@@ -1021,4 +1021,157 @@ class TrailingCommaRuleTest {
                 LintViolation(7, 6, "Missing trailing comma before \")\"")
             ).isFormattedAs(formattedCode)
     }
+
+    @Test
+    fun `Given that a trailing comma is required then add trailing comma after last enum member`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE,
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to true)
+            .hasLintViolation(3, 13, "Missing trailing comma before \"}\"")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given that a trailing comma is is not allowed then remove comma after last enum member`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE,
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to false)
+            .hasLintViolation(3, 13, "Unnecessary trailing comma before \"}\"")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given that a trailing comma is not allowed then nothing happens for enums terminated with semicolon`() {
```

Comment:
I don't understand this use case. The code sample does not contain a traling comma in the enum value list. If it did contain a trailing comma, it should be removed even when the list is terminated with ";".